diff --git a/dev-util/cmake/cmake-3.20.5.ebuild b/dev-util/cmake/cmake-3.20.5.ebuild
index b1cc1455ac6c..1d481a354049 100644
--- a/dev-util/cmake/cmake-3.20.5.ebuild
+++ b/dev-util/cmake/cmake-3.20.5.ebuild
@@ -6,7 +6,7 @@ EAPI=7
 CMAKE_MAKEFILE_GENERATOR="emake" # TODO RunCMake.LinkWhatYouUse fails consistently w/ ninja
 CMAKE_REMOVE_MODULES_LIST=( none )
 inherit bash-completion-r1 cmake elisp-common flag-o-matic multiprocessing \
-	toolchain-funcs virtualx xdg-utils
+	prefix toolchain-funcs virtualx xdg-utils
 
 MY_P="${P/_/-}"
 
@@ -152,6 +152,12 @@ src_prepare() {
 		sed -i -e 's/^#if !defined(_POSIX_C_SOURCE) && !defined(_WIN32) && !defined(__sun)/& \&\& !defined(__APPLE__)/' \
 			Source/cmStandardLexer.h
 	fi
+	if use prefix; then
+		find Modules -name "*.cmake" -type f -print0 | \
+			while IFS= read -r -d '' filename; do
+				hprefixify ${filename}
+			done
+	fi
 
 	# Add gcc libs to the default link paths
 	sed -i \
