--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/include/miopen/solver/implicitgemm_util.hpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/include/miopen/solver/implicitgemm_util.hpp	2021-10-05 11:04:30.770484306 +0800
@@ -48,9 +48,9 @@
 // due to compiler bug, iGEMM xdlops kernels fail verification in some cases, if using "-O3" flag,
 // (but will pass verification with "-O1" flag)
 #define WORKAROUND_SWDEV_251757 1
+// although gfx1031 supports buffer instructions,but it not work properly when we use the
-// although gfx1030 supports buffer instructions,but it not work properly when we use the
 // corresponding llvm intrinsic functions
+// so we disable using those llvm intrinsic functions on gfx1031
-// so we disable using those llvm intrinsic functions on gfx1030
 #define WORKAROUND_MIOPEN_ISSUE_557 1
 
 namespace miopen {
@@ -747,7 +747,7 @@
 {
 #if WORKAROUND_MIOPEN_ISSUE_557
     const auto device_name = ctx.GetStream().GetDeviceName();
+    return !StartsWith(device_name, "gfx1031");
-    return !StartsWith(device_name, "gfx1030");
 #else
     return true;
 #endif
@@ -756,7 +756,7 @@
 static inline bool is_use_v_fmac_f32(const ConvolutionContext& ctx)
 {
     const auto device_name = ctx.GetStream().GetDeviceName();
+    return StartsWith(device_name, "gfx1031");
-    return StartsWith(device_name, "gfx1030");
 }
 
 template <typename T>
@@ -872,7 +872,7 @@
            StartsWith(c.GetStream().GetDeviceName(), "gfx906") ||
            StartsWith(c.GetStream().GetDeviceName(), "gfx908") ||
            StartsWith(c.GetStream().GetDeviceName(), "gfx90a") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx1031");
-           StartsWith(c.GetStream().GetDeviceName(), "gfx1030");
 }
 
 } // namespace solver
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/batchnorm_functions.h	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/batchnorm_functions.h	2021-10-05 11:05:51.652853809 +0800
@@ -143,8 +143,8 @@
 #define MIO_RUNNING_RESULT 0
 #endif
 
+#ifndef MIO_BN_GFX1031
+#define MIO_BN_GFX1031 0
-#ifndef MIO_BN_GFX1030
-#define MIO_BN_GFX1030 0
 #endif
 
 #define UNUSED __attribute__((__unused__))
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/composable_kernel/include/utility/config.hpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/composable_kernel/include/utility/config.hpp	2021-10-05 11:06:08.448515074 +0800
@@ -38,7 +38,7 @@
 #endif
 #endif
 
+// gfx1031 does not support V_MAD/V_MAC,but can use v_fmac_f32
-// gfx1030 does not support V_MAD/V_MAC,but can use v_fmac_f32
 #ifndef CK_USE_AMD_V_FMAC_F32
 #define CK_USE_AMD_V_FMAC_F32 0
 #endif
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivBwdPerAct.cl	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivBwdPerAct.cl	2021-10-05 11:05:51.652853809 +0800
@@ -33,7 +33,7 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
+#if(MIO_BN_GFX1031 == 1 && __AMDGCN__)
-#if(MIO_BN_GFX1030 == 1 && __AMDGCN__)
 #undef __AMDGCN__
 #endif
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivBwdSpatial.cl	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivBwdSpatial.cl	2021-10-05 11:05:51.652853809 +0800
@@ -31,7 +31,7 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
+#if(MIO_BN_GFX1031 == 1 && __AMDGCN__)
-#if(MIO_BN_GFX1030 == 1 && __AMDGCN__)
 #undef __AMDGCN__
 #endif
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl	2021-10-05 11:05:51.652853809 +0800
@@ -32,7 +32,7 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
+#if(MIO_BN_GFX1031 == 1 && __AMDGCN__)
-#if(MIO_BN_GFX1030 == 1 && __AMDGCN__)
 #undef __AMDGCN__
 #endif
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormBwdSpatial.cl	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormBwdSpatial.cl	2021-10-05 11:05:51.652853809 +0800
@@ -32,7 +32,7 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
+#if(MIO_BN_GFX1031 == 1 && __AMDGCN__)
-#if(MIO_BN_GFX1030 == 1 && __AMDGCN__)
 #undef __AMDGCN__
 #endif
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormFwdTrainSpatial.cl	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/kernels/MIOpenBatchNormFwdTrainSpatial.cl	2021-10-05 11:05:51.652853809 +0800
@@ -32,7 +32,7 @@
 #pragma clang diagnostic ignored "-Wsometimes-uninitialized"
 #endif
 
+#if(MIO_BN_GFX1031 == 1 && __AMDGCN__)
-#if(MIO_BN_GFX1030 == 1 && __AMDGCN__)
 #undef __AMDGCN__
 #endif
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/md_graph.cpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/md_graph.cpp	2021-10-05 11:04:30.770484306 +0800
@@ -723,8 +723,8 @@
 
             add_v21_wino("gfx9", {"gfx900", "gfx906", "gfx908"}, 1);
             add_v21_wino("gfx9", {"gfx900", "gfx906", "gfx908"}, 2);
+            add_v21_wino("gfx10", {"gfx1011", "gfx1012", "gfx1031"}, 1);
+            add_v21_wino("gfx10", {"gfx1011", "gfx1012", "gfx1031"}, 2);
-            add_v21_wino("gfx10", {"gfx1011", "gfx1012", "gfx1030"}, 1);
-            add_v21_wino("gfx10", {"gfx1011", "gfx1012", "gfx1030"}, 2);
         }
     }
 
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/ocl/batchnormocl.cpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/ocl/batchnormocl.cpp	2021-10-05 11:05:05.649781341 +0800
@@ -291,8 +291,8 @@
                         " -DMIO_BN_LDSGCN_SIZE=" + std::to_string(ldsgcn) + " -DMIO_BN_N=" +
                         std::to_string(n) + " -DMIO_BN_GRP0=" + std::to_string(xlocalsize) +
                         " -DMIO_BN_GRP1=" + std::to_string(ylocalsize) + " -DMIO_BN_GRP2=" +
+                        std::to_string(zlocalsize) + " -DMIO_BN_GFX1031=" +
+                        ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                        std::to_string(zlocalsize) + " -DMIO_BN_GFX1030=" +
-                        ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
                 if(variant != 4)
                 {
@@ -441,7 +441,7 @@
                     std::to_string(ldsgcn) + " -DMIO_BN_VARIANT=" + std::to_string(variant) +
                     " -DMIO_BN_GRP0=" + std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" +
                     std::to_string(ylocalsize) + " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) +
+                    " -DMIO_BN_GFX1031=" + ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                    " -DMIO_BN_GFX1030=" + ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
                 MIOPEN_LOG_I2(kernel_name << ":: " << parms);
 
@@ -560,8 +560,8 @@
                 " -DMIO_BN_LDS_SIZE=" + std::to_string(ylocalsize) + " -DMIO_BN_GRP0=" +
                 std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" + std::to_string(ylocalsize) +
                 " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) + " -DMIO_BN_NCHW=" +
+                std::to_string(in_nchw) + " -DMIO_BN_GFX1031=" +
+                ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                std::to_string(in_nchw) + " -DMIO_BN_GFX1030=" +
-                ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
             std::string program_name = "MIOpenBatchNormFwdTrainPerAct.cl";
             std::string kernel_name  = "MIOpenBatchNormFwdTrainPerActivation";
@@ -757,7 +757,7 @@
                 " -DMIOPEN_USE_FPMIX=" + std::to_string(static_cast<int>(bfpmixparm)) +
                 " -DMIO_BN_GRP0=" + std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" +
                 std::to_string(ylocalsize) + " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) +
+                " -DMIO_BN_GFX1031=" + ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                " -DMIO_BN_GFX1030=" + ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
             std::vector<size_t> vld;
             std::vector<size_t> vgd;
@@ -1111,8 +1111,8 @@
                         std::to_string(ldsgcn) + " -DMIO_BN_VARIANT=" + std::to_string(variant) +
                         " -DMIO_BN_GRP0=" + std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" +
                         std::to_string(ylocalsize) + " -DMIO_BN_GRP2=" +
+                        std::to_string(zlocalsize) + " -DMIO_BN_GFX1031=" +
+                        ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                        std::to_string(zlocalsize) + " -DMIO_BN_GFX1030=" +
-                        ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
                 }
 
                 MIOPEN_LOG_I2(kernel_name << ":: " << algo_name);
@@ -1227,7 +1227,7 @@
                     std::to_string(ldsgcn) + " -DMIO_BN_VARIANT=" + std::to_string(variant) +
                     " -DMIO_BN_GRP0=" + std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" +
                     std::to_string(ylocalsize) + " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) +
+                    " -DMIO_BN_GFX1031=" + ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                    " -DMIO_BN_GFX1030=" + ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
                 MIOPEN_LOG_I2(kernel_name << ":: " << parms);
 
@@ -1334,8 +1334,8 @@
                 " -DMIO_BN_NCHW=" + std::to_string(in_nchw) + " -DMIO_BN_NGRPS=" +
                 std::to_string(int(std::ceil(float(ygridsize) / ylocalsize))) + " -DMIO_BN_GRP0=" +
                 std::to_string(xlocalsize) + " -DMIO_BN_GRP1=" + std::to_string(ylocalsize) +
+                " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) + " -DMIO_BN_GFX1031=" +
+                ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-                " -DMIO_BN_GRP2=" + std::to_string(zlocalsize) + " -DMIO_BN_GFX1030=" +
-                ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
             if(useSaved)
             {
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/ocl/fusionopbiasbnactivocl.cpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/ocl/fusionopbiasbnactivocl.cpp	2021-10-05 11:05:05.649781341 +0800
@@ -388,7 +388,7 @@
            std::to_string(ldsnogcn) + " -DMIO_BN_LDSGCN_SIZE=" + std::to_string(ldsgcn) +
            " -DMIO_BN_USESAVED=" + std::to_string(static_cast<int>(true)) + " -DMIO_BN_VARIANT=" +
            std::to_string(variant) + " -DMIO_BN_CBA_WRITE_INTERMEDIATE=" + std::to_string(0) +
+           " -DMIO_BN_GFX1031=" + ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-           " -DMIO_BN_GFX1030=" + ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
     compile_config += add;
     MIOPEN_LOG_I2(add);
@@ -599,7 +599,7 @@
            " -DMIOPEN_READ_UNIT=" + std::to_string(read_unit) + " -DMIOPEN_READ_TYPE=" + READ_TYPE +
            " -DMIO_SAVE_MEAN_VARIANCE=" + (saveBatchStats ? "1" : "0") + " -DMIO_RUNNING_RESULT=" +
            ((savePopStats) ? "1" : "0") + " -DMIO_BN_VARIANT=" + std::to_string(variant) +
+           " -DMIO_BN_GFX1031=" + ((handle.GetDeviceName() == "gfx1031") ? "1" : "0");
-           " -DMIO_BN_GFX1030=" + ((handle.GetDeviceName() == "gfx1030") ? "1" : "0");
 
     compile_config += add;
     MIOPEN_LOG_I2(add);
--- /opt/origin/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/target_properties.cpp	2021-07-08 03:16:05.000000000 +0800
+++ /opt/build/portage/sci-libs/miopen-4.3.0/work/MIOpen-rocm-4.3.0/src/target_properties.cpp	2021-10-05 11:08:04.978163431 +0800
@@ -51,7 +51,7 @@
         {"gfx804", "gfx803"},
         {"Vega10", "gfx900"},
         {"gfx901", "gfx900"},
+        {"10.3.0 Navi_flounder 18", "gfx1031"},
-        {"10.3.0 Sienna_Cichlid 18", "gfx1030"},
     };
 
     const char* const p_asciz = miopen::GetStringEnv(MIOPEN_DEBUG_ENFORCE_DEVICE{});
