--- /opt/origin/portage/sci-libs/rocPRIM-4.3.0/work/rocPRIM-rocm-4.3.0/test/rocprim/test_utils_types.hpp	2021-06-08 02:16:47.000000000 +0800
+++ /opt/build/portage/sci-libs/rocPRIM-4.3.0/work/rocPRIM-rocm-4.3.0/test/rocprim/test_utils_types.hpp	2021-09-30 22:30:33.012885698 +0800
@@ -48,7 +48,7 @@
     static constexpr unsigned int block_size = BlockSize;
 };
 
+#if (defined(__gfx1031__))
-#if (defined(__gfx1030__))
     #define warp_param_type(type) \
         warp_params<type, 4U>, \
         warp_params<type, 8U>, \
--- /opt/origin/portage/sci-libs/rocPRIM-4.3.0/work/rocPRIM-rocm-4.3.0/test/rocprim/test_warp_sort.cpp	2021-06-08 02:16:47.000000000 +0800
+++ /opt/build/portage/sci-libs/rocPRIM-4.3.0/work/rocPRIM-rocm-4.3.0/test/rocprim/test_warp_sort.cpp	2021-09-30 22:30:38.372784556 +0800
@@ -34,7 +34,7 @@
     using params = Params;
 };
 
+#if (defined(__gfx1031__))
-#if (defined(__gfx1030__))
     #define warp_sort_param_type(type) \
         warp_params<type, 2U>, \
         warp_params<type, 4U>, \
