diff --color -upr orig/CMakeLists.txt MIOpen-rocm-4.3.0/CMakeLists.txt
--- orig/CMakeLists.txt	2021-10-07 16:18:28.963202825 +0800
+++ MIOpen-rocm-4.3.0/CMakeLists.txt	2021-10-07 16:22:05.470665618 +0800
@@ -69,12 +69,6 @@ set(MIOPEN_ENABLE_SQLITE_BACKOFF On CACH
 option( BUILD_DEV "Build for development only" OFF)
 
 
-# Strip symbols for release
-if(NOT WIN32 AND NOT APPLE)
-    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -s")
-    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -s")
-endif()
-
 rocm_setup_version(VERSION 2.12.0)
 
 list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake )
