Index: src/CMakeLists.txt
===================================================================
--- src.orig/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -305,9 +305,6 @@ install ( TARGETS ${CORE_RUNTIME_TARGET}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT binary
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT binary )
 
-# Install license
-install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.md DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT binary )
-
 # Install public headers
 install ( DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/inc/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hsa COMPONENT dev )
 
