Index: HIP-rocm-5.6.0/tests/catch/CMakeLists.txt
===================================================================
--- HIP-rocm-5.6.0.orig/tests/catch/CMakeLists.txt
+++ HIP-rocm-5.6.0/tests/catch/CMakeLists.txt
@@ -156,7 +156,7 @@ if (WIN32)
 endif()
 
 if(HIP_PLATFORM MATCHES "amd" AND HIP_COMPILER MATCHES "clang")
-    add_compile_options(-Wall -Wextra -pedantic -Werror -Wno-deprecated)
+    add_compile_options(-Wall -Wextra -pedantic -Wno-deprecated)
 endif()
 
 cmake_policy(PUSH)
