Index: hipamd/packaging/CMakeLists.txt
===================================================================
--- hipamd.orig/packaging/CMakeLists.txt
+++ hipamd/packaging/CMakeLists.txt
@@ -60,7 +60,6 @@ if(HIP_PLATFORM STREQUAL "amd" )
 
 #TODO:This do not belong in BINARY package.
 #Keeping it as is for now
-install(FILES ${CMAKE_BINARY_DIR}/hipamd/.hipInfo DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT binary)
 
 install(FILES ${CMAKE_BINARY_DIR}/hipamd/hip-config.cmake ${CMAKE_BINARY_DIR}/hipamd/hip-config-version.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/hip COMPONENT binary)
 install ( EXPORT hip-targets  FILE hip-targets.cmake NAMESPACE hip:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/hip  COMPONENT binary)
