Index: hipamd-rocm-5.2.0/CMakeLists.txt
===================================================================
--- hipamd-rocm-5.2.0.orig/CMakeLists.txt
+++ hipamd-rocm-5.2.0/CMakeLists.txt
@@ -395,7 +395,7 @@ if(NOT ${INSTALL_SOURCE} EQUAL 0)
     if(WIN32)
       install(DIRECTORY ${HIP_COMMON_DIR}/cmake DESTINATION lib64)
     else()
-      install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION CONFIG_PACKAGE_INSTALL_DIR)
+      install(DIRECTORY ${HIP_COMMON_DIR}/cmake/ DESTINATION ${CONFIG_PACKAGE_INSTALL_DIR})
     endif()
 endif()
 
