Index: HIP-rocm-5.1.3/bin/hipcc.pl
===================================================================
--- HIP-rocm-5.1.3.orig/bin/hipcc.pl
+++ HIP-rocm-5.1.3/bin/hipcc.pl
@@ -105,8 +105,12 @@ $ROCM_PATH      =   $hipvars::ROCM_PATH;
 $HIP_VERSION    =   $hipvars::HIP_VERSION;
 $HSA_PATH       =   $hipvars::HSA_PATH;
 $HIP_ROCCLR_HOME =   $hipvars::HIP_ROCCLR_HOME;
+$HIP_INCLUDE_PATH =   $hipvars::HIP_INCLUDE_PATH;
+$DEVICE_LIB_PATH =   $hipvars::DEVICE_LIB_PATH;
+$HIP_LIB_PATH =   $hipvars::HIP_LIB_PATH;
+$HIP_CLANG_INCLUDE_PATH =   $hipvars::HIP_CLANG_INCLUDE_PATH;
 
-if ($HIP_PLATFORM eq "amd") {
+if ($HIP_PLATFORM eq "NotForGentoo") {
   # If using ROCclr runtime, need to find HIP_ROCCLR_HOME
   if (!defined $DEVICE_LIB_PATH and -e "$HIP_ROCCLR_HOME/lib/bitcode") {
     $DEVICE_LIB_PATH = "$HIP_ROCCLR_HOME/lib/bitcode";
