Index: rocm_smi_lib-rocm-6.0.0/CMakeLists.txt
===================================================================
--- rocm_smi_lib-rocm-6.0.0.orig/CMakeLists.txt
+++ rocm_smi_lib-rocm-6.0.0/CMakeLists.txt
@@ -271,15 +271,7 @@ install(EXPORT rocm_smiTargets
 
 #License file
 set(CPACK_RPM_PACKAGE_LICENSE "NCSA")
-# install license file in share/doc/rocm_smi-asan folder
-if( ENABLE_ASAN_PACKAGING )
-  install(FILES ${CPACK_RESOURCE_FILE_LICENSE}
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/doc/${ROCM_SMI}-asan RENAME LICENSE.txt
-        COMPONENT asan)
-endif()
-install( FILES ${CPACK_RESOURCE_FILE_LICENSE}
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/doc/${ROCM_SMI} RENAME LICENSE.txt
-        COMPONENT dev)
+
 
 ###########################
 # Packaging directives
